### ISSUE TITLE
Bitrunner avatars can gain antagonist data (in this case, to murder their main mob)

### ISSUE THREAD

**Issue by iprice** [tgstation/tgstation#78618]
Reporting client version: 514.1589

## Round ID:
[215373](https://scrubby.melonmesa.com/round/215373)

## Testmerges:
- [ArCargo: Adds the Galactic Materials Stock Market V1.2 (Free Market Edition)](https://www.github.com/tgstation/tgstation/pull/78500)
- [TTS: Gas Mask muffling, Hailer Mask voice effects, support for more filters that use samplerate, voice effects for lizards, ethereals, and xenomorphs.](https://www.github.com/tgstation/tgstation/pull/78567)

## Reproduction:

Player as a bitrunner avatar gained antag status, with a primary objective of killing their main toon 
The antag status does not transfer back to the main character, instead leaving a no-client traitor on the antags panel when the bit avatar returns 'home', and eventually "Body destroyed" once the domain is ended.

```
[2023-09-26 03:24:24.278] DYNAMIC: MediaNocheUwU/(Karenza Ali) was selected by the Syndicate Sleeper Agent ruleset and has been made into a midround traitor.
```
![image](https://github.com/tgstation/tgstation/assets/3789327/1d737ab3-dc21-401c-b226-745862db3876)

![image](https://github.com/tgstation/tgstation/assets/3789327/d34d4007-0ec1-44be-a123-dde79e69400a)

**Comment by jlsnow301** [comment]
This should have been fixed in #78530, I'll look further into it if it's still going on

**Comment by jlsnow301** [comment]
Closing this as fixed, reopen if it isn't
